src/core/meshcop/border_agent.cpp

```diff
@@ -453,8 +453,7 @@ void BorderAgent::HandleRelayReceive(const Coap::Message &aMessage)
     Coap::Message *message = NULL;
     otError        error;
 
-    VerifyOrExit(aMessage.GetType() == OT_COAP_TYPE_NON_CONFIRMABLE && aMessage.GetCode() == OT_COAP_CODE_POST,
-                 error = OT_ERROR_DROP);
+    VerifyOrExit(aMessage.IsNonConfirmable() && aMessage.GetCode() == OT_COAP_CODE_POST, error = OT_ERROR_DROP);
```

Comment:
nit: Add `IsPost()` too?

Reply:
This PR is using the existing helper methods in `Coap::Message` that check the CoAP "type". The `POST` is a CoAP "code" (not a "type"), and we currently don't have helper methods like `IsPost()`. 

There are only 4 CoAP types and we already did have helper methods defined for all 4 types, but there are many more CoAP codes so I am not sure if we want helper methods for checking the code values.